semgrep_output_v1.atd

```diff
@@ -801,13 +801,48 @@ type found_dependency <ocaml attr="deriving show"> = {
   ?line_number: int option;
 }
 
+
+(* json names are to maintain backwards compatibility with the python enum it is replacing *)
+type sca_parser_name <ocaml attr="deriving show"> = [
+  | Gemfile_lock <json name="gemfile_lock">
+  | Go_mod <json name="go_mod">
+  | Go_sum <json name="go_sum">
+  | Gradle_lockfile <json name="gradle_lockfile">
+  | Gradle_build <json name="gradle_build">
+  | Jsondoc <json name="jsondoc">
+  | Pipfile <json name="pipfile">
+  | Pnpm_lock <json name="pnpm_lock">
+  | Poetry_lock <json name="poetry_lock">
+  | Pyproject_toml <json name="pyproject_toml">
+  | Requirements <json name="requirements">
+  | Yarn_1 <json name="yarn_1">
+  | Yarn_2 <json name="yarn_2">
+  | Pomtree <json name="pomtree">
+  | Cargo_parser <json name="cargo">
+  | Composer_lock <json name="composer_lock">
+]
+
+
+
+type dependency_parser_error <ocaml attr="deriving show"> = {
+  path: string;
+  parser: sca_parser_name;
+  reason: string;
+  (* Not using `position` because this type must be backwards compatible with the python class it is replacing *)
```

Comment:
We can refactor the python code, and I agree we should. The main difficulty is with supporting old versions of the CLI. We have no choice but to support this type for a while. 

Would be really cool if ATD had some kind of "compatibility sum type", where I could say "it's either this or this, and ATD should insert the tag depending on the shape of the data". Like

```
type sca_parser_error <compat-sum> = [
  | OldType of old_type
  | NewType of new_type
]
```

And then the actual data is untagged, since old versions would just be sending `old_type` by itself